File: StudyKotlin/app/src/main/java/com/lily/kotlin/singleton/java/InnerSingleton.java
package com.lily.kotlin.singleton.java;

/**
 * @Author rape flower
 * @Date 2018-06-29 14:20
 * @Describe 单例模式：Java方式
 * <p>内部类式</p>
 */
public class InnerSingleton {

    private InnerSingleton() {

    }

    private static class Holder {
        private static InnerSingleton instance = new InnerSingleton();
    }

    public static InnerSingleton getInstance() {
        return Holder.instance;
    }
}

File: StudyKotlin/app/src/main/java/com/lily/kotlin/singleton/kotlin/InnerSingleton.kt
package com.lily.kotlin.singleton.kotlin

/**
 * @Author rape flower
 * @Date 2018-06-29 14:29
 * @Describe 单例模式：Kotlin方式
 * <p>内部类式</p>
 */
class InnerSingleton private constructor() {

    private object Holder {
        val INSTANCE = InnerSingleton()
    }

    companion object {
        fun getInstance(): InnerSingleton {
            return Holder.INSTANCE
        }
    }
}